### ISSUE TITLE
DRO task oddities

### ISSUE THREAD

**Issue by jalepisto** [ibm-mas/ansible-devops#1304]
DRO task sets redhat-marketplace-pull-secret value to be ibm-entitlement key. Can't be right. At least its complaining 
"failed to unmarshal secret: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type v1.Secret'"

Another problem with this forced Red Hat Marketplace setup is, check the image.  Why is it forced whenits not available everywhere.
![image](https://github.com/ibm-mas/ansible-devops/assets/123641253/aa719b46-0d5d-418a-a02b-80baa925cbd9)

**Comment by lokesh-sreedhara** [comment]
@jalepisto can u provide logs from the ansible tasks? i would like to see at what step this error occurs. also can you confirm the IBM entitlement key used here was of the correct format without any whitespaces?

**Comment by dacleyra** [comment]
If you get the key from either (must me logged in) 
https://swc.saas.ibm.com/en-us/redhat-marketplace/account/keys
https://swc.saas.ibm.com/en-us/software-central/account/keys
Create it as name: `redhat-marketplace-pull-secret`

If you get the key from here https://myibm.ibm.com/products-services/containerlibrary
Create it as name: `ibm-entitlement-key`

Data Reporter Operator and IBM Metrics Operator container images are hosted on quay.io


You can check the ibm-metrics-operator logs to ensure your key is decoded correctly
```
oc -n redhat-marketplace logs $(oc -n redhat-marketplace get pod -l redhat.marketplace.com/name=ibm-metrics-operator -o jsonpath='{.items..metadata.name}')
```

And check the MarketPlaceConfig Status for any other warnings
```
oc -n redhat-marketplace describe marketplaceconfig
```

**Comment by jalepisto** [comment]
I have linked an account to Redhat Marketplace and created/downloaded a pull secret. As it has a pull secret of its own I don't get how IBM pull secret is supposed to work. I'd be tempted to say it didn't. Now it seems to work with RH marketplace pull secret.

The secret still has  annotations:
    marketplace.redhat.com/rhm-operator-secret-message: 'failed to unmarshal secret: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type v1.Secret'
    marketplace.redhat.com/rhm-operator-secret-status: error

which where there with the original IBM pull secret. I just replaced it with the pull secret from RH marketplace.

If you go to website of RH Marketplace it tells you its not available for all countries (Finland and Sweden included that I am concerned,) So how come this is forced,

Marketplace conditions are good except registration.  I don't know  what that registration is referring to. OCP is registered.
Complete
True
14 May 2024, 08:12
FinishedInstall
Finished Installing necessary components
Installing
False
14 May 2024, 08:12
FinishedInstall
Finished Installing necessary components
RHMAccountExists
True
14 May 2024, 08:12
RHMAccountExists
RHM/Software Central account exists
Registered
False
14 May 2024, 08:12
ClusterRegistered
Cluster registration pending: UNREGISTERED

-janne
________________________________
